docs/faq.rst

```diff
@@ -685,6 +687,33 @@ The above will block flagged IPs for a week, you can of course change it to you
 
 Using DOCKER-USER chain ensures that the blocked IPs are processed in the correct order with Docker. See more in: https://docs.docker.com/network/iptables/
 
+Option 2:  Use ipset together with iptables
+IMPORTANT: You have to install ipset on the host system, eg. `apt-get install ipset` on a Debian/Ubuntu system.
+
+See ipset homepage for details on ipset, https://ipset.netfilter.org/.
+
+ipset and iptables provide one big advantage over just using iptables: This setup reduces the overall iptable rules.
+There is just one rule for the bad authentications and the IPs are within the ipset. 
+Specially in larger setups with a high amount of brute force attacks this comes in handy.
+Using iptables with ipset might reduce the system load in such attacks significantly.
+
+.. code-block:: bash
+
+  [Definition]
+
+  actionstart = actionstart = ipset --create f2b-bad-auth iphash
+                iptables -I DOCKER-USER -p tcp -m multiport --dports 1:1024 -m set --match-set f2b-bad-auth src -j DROP
+
+  actionstop = iptables -D DOCKER-USER -p tcp -m multiport --dports 1:1024 -m set --match-set f2b-bad-auth src -j DROP
+               ipset --destroy f2b-bad-auth
+
+
+  actionban = ipset --test f2b-bad-auth <ip> ||  ipset --add f2b-bad-auth <ip>
+
+  actionunban = ipset --test f2b-bad-auth <ip> && ipset --del f2b-bad-auth <ip>
```

Comment:
Why do you run a test first? The man page says that the command will be ignored if the entry is already there (or missing)

Reply:
You'll get an exit code `1` and some "error" message, eg.:
```
# ipset --add f2b-bad-auth 37.x.x.x
ipset v7.10: Element cannot be added to the set: it's already added
# echo $?
1
``` 
ipset is ignoring it, but still being particular about it. 
That's why I thought it would be less "noisy" be doing checks upfront.

Reply:
The right syntax is ``ipset add f2b-bad-auth <ip> -exist`` ... like that there is no error thrown if it's removed/deleted twice.

```
Synopsis
ipset [ OPTIONS ] COMMAND [ COMMAND-OPTIONS ]

COMMANDS := { create | add | del | test | destroy | list | save | restore | flush | rename | swap | help | version | - }

OPTIONS := { -exist | -output { plain | save | xml } | -quiet | -resolve | -sorted | -name | -terse } 
```